Тайпан v2 (Экономика + Квест система для карго)

Furukia opened the pull request:
## Основная задумка
Вместо стандартной системы с получением в карго денег через плазму диски и т.д. Карго Тайпана будет получать "Контракты" различные случайно генерируемые задачи, за выполнение которых они будут зарабатывать деньги и другие награды. Большинство задач нацелено на производство чего либо конкретного и отправку этого синдикату, для выполнения контракта. 

Так же эта система нацелена на введение и тестирование новой экономики, пока что на тайпане, уже позже в изменённой итерации на станции.  Для этого планируется система цен, сканеры для оценивания предметов. Средняя цена для почти каждого существующего предмета, опять же интеграция этого в систему квестов и т.д.

## Дополнительные мелочи
Чёрно позолоченный стиль интерфейса синдиката для карго тайпана, системы квестов, аплинка контрактника и простого аплинка, если его владелец - контрактник.

## Чеклист Квест Системы

- [x] Базовый костяк в виде Датума "quest system" и самих "quest"-ов.
- [ ] Создание и наполнение weight - листов для генерации квестов
-  - [x] Листы для квеста на сборку Меха
-  - [x] Листы для квеста на создание Вируса
-  - [x] Листы для квеста на создание растения с особыми трейтами
-  - [x] Листы для квеста на создание гранаты с определённым набором химикатов и корпусом
-  - [x] Листы для квеста на доставку определённых органов/частей тел
-  - [x] Листы для квеста на шприцы с определёнными генами
-  - [ ] Листы для квеста на маленьких ботов
-  - [x] Листы для квеста на создание оружия
- [x] Возможно - перевод листов в глобальные листы, вместо части "quest system"
- [ ] Генерация всех задуманных видов квестов
-  - [x] Генерация квеста на сборку Меха
-  - [x] Генерация квеста на создание Вируса
-  - [x] Генерация квеста на создание растения с особыми трейтами
-  - [x] Генерация квеста на создание гранаты с определённым набором химикатов и корпусом
-  - [x] Генерация квеста на доставку определённых органов/частей тел
-  - [x] Генерация квеста на шприцы с определёнными генами
-  - - [ ] Возможно добавить "мартышек" как усложнённый вариант для переноски генов вместо шприцов из аппарата.
-  - [ ] Генерация квеста на маленьких ботов
-  - [ ] Генерация квеста на создание оружия
- [ ] Код для генерации наград за квесты
- [ ] Код для проверки условий выполнения квеста
- [ ] Код для создания Кастомных квестов админами
- [ ] Капсула сканер с областью 3x3 перед собой, для собирания и компоновки заказанного квестом. 
- [ ] Отдельная админская консоль для отлеживания существующих "чёрных рынков" и их квест систем
- [ ] Возможно добавление некой платы вставляемой в консоли синдикарго для добавления этой квест системы к ним.
- [ ] Возможно стоит уделить внимание добавлению отдельного investigate логирования для всей системы квестов
- [ ] Интерфейс TGUI...
-  - [ ]  для окна квестов
-  - [ ]  для окна админского создания квестов
-  - [ ]  для капсулы

## Чеклист Системы Экономики
- [ ] Заметка для себя: Изучить вот этот ПР от Оффов https://github.com/ParadiseSS13/Paradise/pull/19209 и почерпнуть оттуда часть идей
- [ ] Выдача средней цены всем предметам
- [ ] Код который будет использовать эту среднюю цену и вызываться при инициализации предмета, для определения его стоимости в виде случайного отклонения от среднего значения
- [ ] Инструмент сканер способный оценить любой предмет и показать цену игроку.
- [ ] Прикручивание денежных счетов для карго и возможно - отдельные скрытые счета для Тайпана.
- [ ] Доработка торговли с агентами в виде возможности тайпану выставлять товары с АВТОМАТИЧЕСКИМ определением их цены игрою. А для трейторов - покупать их за кредиты, либо через аплинк, либо через некий планшетик покупаемый в аплинке.
- [ ] To be continued

Furukia committed:
Начало переработки карго.

* Генератор квестов на создание мехов
* Генератор квестов на создание вирусов
* Генератор квестов на создание гранат(сыроватый пока)
* Новый стиль синди интерфейса и добавление его контрактнику и в карго.

Furukia committed:
Ещё пару генераторов, недопиленная система и т.д.

лень писать описание. Структурирую все изменения потом в последних комитах. И ПРе

Furukia committed:
Перепил генерации сложности и её влияния на квесты

Sirryan2002 commented:
Если у вас есть какие-либо вопросы об упомянутом PR экономики, не стесняйтесь написать мне в dm/ping.

Furukia commented:
>Если у вас есть какие-либо вопросы об упомянутом PR экономики, не стесняйтесь написать мне в dm/ping.

Hi, for now i just want to read ur PR and it's code carefully and maybe copy some ideas. I want to create a quest cargo system and economic system for our russian paradise fork. Specifically for cargo on our syndicate space station "RAMSS Taipan" (ghost role) initially created by me. And later if it will work out, adopt those "quest" features to the main cargo. It's essencially a black market system, where players will get requests from the market to create and give them something. In return for money and other rewards. And while i do that, i want to make our economy better, and your PR was recommended to me, cause it does exactly that.

Furukia committed:
Лист органов и ещё пару мелочей

* Лист с органами для квеста на них
* Небольшое изменение Цифр сложности квестов
* tgui
* Небольшой фикс карты тайпана.

Furukia committed:
Генерация квестов на органы + дефайны для типов квестов

* Генерация квестов на органы и части тела
* Пересчёт веса в листе с органами
* Дефайны для всех типов квестов
* Ребилд TGUI бандла после ребейза

Furukia committed:
Гены

* Лист генов для квестов на шприцы с ними
* Генерация этого вида квестов

Furukia committed:
Уборка

* Разбила файл квест системы на тематические файлы для разных квестов. В каждом из которых должны быть:
- глобальные листы для генерации квестов
- прок генерации определённого вида квеста
- прок проверки выполнения определённого типа квеста
- другие возможные проки для нужного типа квеста

* Сделала листы для генерации - Глобальными

Furukia committed:
Redspace Packager - начало

Furukia committed:
Лист оружия и имплантов
